From e70da09371afafd93a683bfb475348ba5fecf59e Mon Sep 17 00:00:00 2001
From: Thomas Ramon <ramon.thomas.74@gmail.com>
Date: Fri, 23 Feb 2024 06:41:26 +0100
Subject: [PATCH] fix: 6.7.5-200.fc39.x86_64 compilation error
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This should fix this error when compiling :

/tmp/evdi/module/evdi_gem.c: In function ‘evdi_gem_fault’:
/tmp/evdi/module/evdi_gem.c:214:40: error: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘loff_t’ {aka ‘long long int’} [-Werror=sign-compare]
  214 |         if (!obj->pages || page_offset >= num_pages)
      |                                        ^~
---
 module/evdi_gem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/module/evdi_gem.c b/module/evdi_gem.c
index efae6e1..6298d1b 100644
--- a/module/evdi_gem.c
+++ b/module/evdi_gem.c
@@ -211,7 +211,7 @@ int evdi_gem_fault(struct vm_fault *vmf)
 
 	page_offset = (vmf->address - vma->vm_start) >> PAGE_SHIFT;
 
-	if (!obj->pages || page_offset >= num_pages)
+	if (!obj->pages || (unsigned)page_offset >= num_pages)
 		return VM_FAULT_SIGBUS;
 
 	page = obj->pages[page_offset];
